src/modules/init_maps.rs

```diff
@@ -133,12 +131,14 @@ fn map_storage_changes(blk: eth::Block) -> Result<StorageChanges, Error> {
                     }
                 }
             }
-            if let Some(event) =
-                abi::rewards_manager::events::RewardsAssigned::match_and_decode(&log)
-            {
+            if let Some(event) = abi::staking::events::AllocationClosed::match_and_decode(&log) {
+                // We track the AllocationClosed event instead of RewardsAssigned here. I think it is because their calls are different
+                // when their transaction is the same. And we have the call data here, so we need to use the event that shares the same call
+                // with the storage_change
                 for storage_change in &call_view.call.storage_changes {
                     if storage_change.address.eq(&STAKING_CONTRACT) {
                         if storage_change.key == utils::find_key(&event.indexer, 20, 2) {
+                            //trx.hash is not a great id since multiple allocations might be closed in one transaction. Should be updated
```

Comment:
That would be unique but in the case multiple allocations are closed in one transaction, we need a way to tell which rewards_assigned is associated with which delegation_pool. Ordinal does not help with it since everything will have a different ordinal. I haven't yet come up with a solution

Reply:
Allocation ID would be unique (either coupled with the tx hash or not), but I'm not sure if it's available when pairing it with the delegation pool 🤔

Reply:
Makes sense, thanks a lot for the hint! Implementing it now

Reply:
I think it should be resolved now

Reply:
Looks good, and it's awesome that we are not restricted like in subgraphs, where we actually need to make IDs predictable 😅, here we can simply look up the ones we care about and compare other fields, love it ;)